core/src/modules/ops/record.ts

```diff
@@ -1105,54 +1087,20 @@ export namespace RecordOps {
       }
     );
 
-    await completeRecordImports(
-      records.map((p) => p.id),
-      toExport
+    await Import.update(
+      { state: "complete", importedAt: now },
+      {
+        where: {
+          recordId: { [Op.in]: records.map((p) => p.id) },
+          state: "importing",
+          recordAssociatedAt: { [Op.lt]: now },
+        },
+      }
     );
```

Comment:
There's a possible race condition here where a new import may have appeared in the middle of this method, the record should have been moved to 'pending' again.  BUT, we looked up only those records who had no pending properties a few lines up - the transactional safety of this method should make this OK